export_offline.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime
+import os
+
+import shutil
+from PyQt4 import QtCore
+from qgis.core import QgsMapLayerRegistry, QgsProject, QgsOfflineEditing, QgsRasterLayer
+
+from .file_utils import fileparts
+from .config import OFFLINE
+from .data_source_utils import  SHP_EXTENSIONS,change_layer_data_source, \
+    project_get_always_offline_layers
+
+def get_layer_ids_to_offline_convert( remote_layers, remote_save_mode):
+    layer_ids = []
+    if remote_save_mode == OFFLINE:
+        for layer in remote_layers:
+            layer_ids.append(layer.id())
+
+    for layer in project_get_always_offline_layers():
+        layer_ids.append(layer.id())
+    return layer_ids
+
+
+def handle_shpfiles( dataPath, shpfile_layers):
+    for shpfile_layer in shpfile_layers:
+        shpfile = shpfile_layer.source()
+        parent, fn, ext = fileparts(shpfile)
+        new_file_path = os.path.join(dataPath, fn + ext)
+        # copy surrounding files as well
+        for ext in SHP_EXTENSIONS:
+            file_path = os.path.join(parent, fn + ext)
+            if os.path.exists(file_path):
+                shutil.copy(file_path, dataPath)
+        change_layer_data_source(shpfile_layer, new_file_path)
+
+
+def handle_rasters( dataPath, raster_layers):
+    for raster_layer in raster_layers:
+        file_path = raster_layer.source()
+        new_file_path = os.path.join(dataPath, os.path.basename(file_path))
+        shutil.copy(file_path, new_file_path)
+        change_layer_data_source(raster_layer, new_file_path)
```

Comment:
Converting to offline changes the project, the user should not save (ie
overwrite) when the project has been converted to offline, but use the
exported project in the folder that is opened. Does this address your
concern @mbernasconi?

Στις 4 Αυγ 2016 3:09 ΜΜ, ο χρήστης "Marco Bernasocchi" <
notifications@github.com> έγραψε:

> In export_offline.py
> <https://github.com/opengisch/QFieldSync/pull/1#discussion_r73517906>:
>
> > +        parent, fn, ext = fileparts(shpfile)
> > +        new_file_path = os.path.join(dataPath, fn + ext)
> > +        # copy surrounding files as well
> > +        for ext in SHP_EXTENSIONS:
> > +            file_path = os.path.join(parent, fn + ext)
> > +            if os.path.exists(file_path):
> > +                shutil.copy(file_path, dataPath)
> > +        change_layer_data_source(shpfile_layer, new_file_path)
> > +
> > +
> > +def handle_rasters( dataPath, raster_layers):
> > +    for raster_layer in raster_layers:
> > +        file_path = raster_layer.source()
> > +        new_file_path = os.path.join(dataPath, os.path.basename(file_path))
> > +        shutil.copy(file_path, new_file_path)
> > +        change_layer_data_source(raster_layer, new_file_path)
>
> won't this change the original project?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/opengisch/QFieldSync/pull/1/files/fcbce41e2118ef3d7e54ce576478a0600d0bf272#r73517906>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA7VK6t5FZZV55kNjuN4aR4Zk6aBu_0nks5qceR_gaJpZM4IpZ7z>
> .
>